src/components/confirm2FA/confirm2FA.tsx

```diff
@@ -0,0 +1,156 @@
+  const [disabledContinueButton, setDisabledContinueButton] = useState(true);
+
+  const Lost2FAModal = (props: any) => {
+    const [disabledNew2FAButton, setDisabledNew2FAButton] = useState(true);
+    return (
+      <Modal show={showModal} size="lg" onHide={() => setShowModal(false)}>
+        <Modal.Header>
+          <Modal.Title>Create new setup for authenticator app</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p>
+            In the case that you lost your phone or the setup of your
+            authenticator app, you can create a new authentication code setup.
+          </p>
+          <p className="fw-bold">
+            However, all contact addresses that had been verified before will
+            need to be verified again
+          </p>
+
+          <div>
+            <input
+              type="checkbox"
+              id="agree"
+              className="me-2"
+              onChange={(e) => {
+                setDisabledNew2FAButton(!e.target.checked);
+                console.log(e.target.checked);
+              }}
+            />
+            <label htmlFor="agree" className="d-inline">
+              I acknowledge that all previous contact addresses will require
+              re-verification if a new authenticator setup is requested
+            </label>
+          </div>
+          <br />
+          <Button
+            disabled={disabledNew2FAButton}
+            onClick={() => {
+              navigate("/setup-2fa");
+            }}
+          >
+            <FontAwesomeIcon icon={faCircleArrowRight} />
+            &nbsp; Continue
+          </Button>
+          <Button
+            variant="gray"
+            className="ms-2 text-white"
+            onClick={() => setShowModal(false)}
+          >
+            Cancel
+          </Button>
+        </Modal.Body>
+      </Modal>
+    );
+  };
+
+  return (
+    <Container className="mt-4">
+      <h2>Two-factor authentication</h2>
+      <Lost2FAModal />
+      <div className="w-100">
+        <p>
+          Please enter the 6-digit authentication code generated by your
+          authenticator app
+        </p>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            if (inputted2FA === "123456") navigate("/");
+            else setShowError(true);
+          }}
+        >
+          <OverlayTrigger
+            placement="right"
+            overlay={<Tooltip>Invalid code entered</Tooltip>}
+            rootClose={true}
+            show={showError}
+          >
+            <input
+              type="text"
+              className="text-center fs-2"
+              style={{ letterSpacing: "0.5em", paddingLeft: "0.5em" }}
+              required
+              minLength={6}
+              maxLength={6}
+              pattern="[0-9]{6}"
+              size={6}
+              onKeyDown={(e) => {
+                if (
+                  e.key.match("^[a-zA-Z]{1}$") &&
```

Comment:
```suggestion
                  !e.key.match("^[0-9]{1}$") &&
```

Wouldn't that be better?

Reply:
Made it similar but differently. I was meant to deal with special characters but it completely escaped my mind

Reply:
Yes, that's probably better to allow for cursor movement etc.